src/main/java/org/reactome/release/cosmicupdate/Main.java

```diff
@@ -0,0 +1,337 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.reactome.release.common.ReleaseStep;
+import org.reactome.release.common.dataretrieval.cosmic.COSMICFileRetriever;
+import org.reactome.util.general.GUnzipCallable;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+public class Main extends ReleaseStep
+{
+	@Parameter(names = {"-u"}, description = "Specifies that the updates should be performed.")
+	private boolean executeUpdate;
+	
+	@Parameter(names = {"-d"}, converter = DurationConverter.class,
+			description = "Download the files, if the age of local files exceeds the specified max age, or if the files don't exist."
+					+ "Omitting this parameter means download will not occur. Specifying -d with a 0 value will force a download."
+					+ "Format for a Duration can be found here: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-"
+					+ " Example: PT48H == \"48 hours\"")
+	private Duration fileAge;
+	
+	@Parameter(names = {"-c"}, description = "The path to the configuration file. Default is src/main/resources/config.properties")
+	private String configPath = "src/main/resources/config.properties";
+	
+//	@Parameter(names = {"-a"}, description = "The path to the authorization file, containing usernames and passwords. Default is src/main/resources/auth.properties")
+//	private static String authPath = "src/main/resources/auth.properties";
+	
+	private static String COSMICMutantExport;
+	private static String COSMICFusionExport;
+	private static String COSMICMutationTracking;
+
+	private static String COSMICMutantExportURL;
+	private static String COSMICFusionExportURL;
+	private static String COSMICMutationTrackingURL;
+	
+	private static String COSMICUsername;
+	private static String COSMICPassword;
+	
+	private static final Logger logger = LogManager.getLogger();
+	private static long personId;
+
+	public static void main(String... args)
+	{
+		try
+		{
+			Main cosmicUpdateStep = new Main();
+			JCommander.newBuilder().addObject(cosmicUpdateStep).build().parse(args);
+			Properties configProps = cosmicUpdateStep.getConfig();
+			cosmicUpdateStep.executeStep(configProps);
+		}
+		catch (SQLException e)
+		{
+			logger.error("Error with SQL Connection", e);
+			e.printStackTrace();
+		} catch (Exception e) {
+			logger.error("Exception caught", e);
+		}
+		logger.info("All done.");
+		
+	}
+
+	private Properties getConfig()
+	{
+		boolean noConfigSpecified = this.configPath != null && this.configPath.trim().isEmpty();
+//		boolean noAuthConfigSpecified = Main.authPath != null && !Main.authPath.trim().isEmpty();
+		if (!noConfigSpecified /* && !noAuthConfigSpecified */)
```

Comment:
Would this condition need to be uncommented for a future code change?  If so, maybe a TODO block to detail the circumstances where it would be used?

Reply:
My plan was to have authentication values come from a separate config file. Feature is not yet complete.